bodhi/tests/server/consumers/test_pungi_wrapper.py

```diff
@@ -0,0 +1,419 @@
+                      just_phases=[], old_composes=None, koji_event=None, supported=False,
+                      logger=logger, notifier=mock.Mock())
+
+    return compose
+
+
+@pytest.fixture(scope="module")
+def get_pungi_conf(tmpdir, config_str=None):
+    conf = tmpdir.mkdir("conf").join("pungi.conf")
+    config_data = """ # RELEASE
+    release_name = "Fedora"
+    release_short = "Fedora"
+    release_version = "23"
+
+    # GENERAL SETTINGS
+    comps_file = "comps-f23.xml"
+    variants_file = "variants-f23.xml"
+
+    # KOJI
+    koji_profile = "koji"
+    runroot = False
+
+    # PKGSET
+    sigkeys = [None]
+    pkgset_source = "koji"
+    pkgset_koji_tag = "f23"
+
+    # CREATEREPO
+    createrepo_checksum = "sha256"
+
+    # GATHER
+    gather_source = "comps"
+    gather_method = "deps"
+    greedy_method = "build"
+    check_deps = False
+
+    # BUILDINSTALL
+    bootable = True
+    buildinstall_method = "lorax"
+    release_is_layered = True
+
+    base_product_name = "Fedora"
+    base_product_short = "Fedora"
+    base_product_version = "23" """
+    if config_str:
+        config_data = config_str
+    conf.write(config_data)
+
+    return conf
+
+
+@pytest.mark.usefixture("get_pungi_conf")
+class TestPungiConfig(object):
+
+    def test_load_config_from_file(self, tmpdir):
+        logger = mock.Mock()
+        pungi_conf = get_pungi_conf(tmpdir)
+
+        conf_xml = PungiConfig(path=str(pungi_conf), logger=logger)
+        assert isinstance(conf_xml, dict)
+        assert isinstance(conf_xml["sigkeys"], list)
+        assert conf_xml["release_name"] == "Fedora"
+        assert conf_xml["release_version"] == "23"
+
+    def test_valid_pungi_config(self, tmpdir):
+        logger = mock.Mock()
+        config_data = """ # RELEASE
+        release_name = "Fedora"
+        release_short = "Fedora"
+        release_version = "23" """
+        pungi_conf = get_pungi_conf(tmpdir, config_data)
+
+        with pytest.raises(Exception):
+            PungiConfig(path=str(pungi_conf), logger=logger)
+
+
+@pytest.fixture(scope="module")
+def get_updates_and_builds():
+    updates = [u'host-master-20170830200108', u'platform-master-20170818100407']
+    builds = [
+        u'platform-master-20160818100407',
+        u'platform-master-20160818100408',
+        u'shim-master-20160502110605',
+        u'shim-master-20160502110601',
+        u'host-master-20170830200108',
+        u'host-master-20170830200109',
+        u'installer-master-20170822180922',
+        u'installer-master-20170822180920'
+    ]
+    mock_updates = []
+    mock_builds = []
+    for update in updates:
+        mock_update = mock.Mock()
+        mock_update.title = update
+        mock_updates.append(mock_update)
+
+    for build in builds:
+        mock_build = mock.Mock()
+        mock_build.nvr = build
+        mock_builds.append(mock_build)
+
+    return mock_updates, mock_builds
```

Comment:
Why not use the ```BaseTest``` class and use real Update obects? This will be a far more useful test than a mock will be.